apps/backend/src/modules/users/user.entity.ts

```diff
@@ -1,26 +1,28 @@
 import { type Entity } from "~/libs/types/types.js";
+import { type GroupModel } from "~/modules/groups/group.model.js";
+import { type PermissionModel } from "~/modules/permissions/permission.model.js";
 
 import { type UserAuthResponseDto } from "./libs/types/types.js";
 
 class UserEntity implements Entity {
 	private createdAt: null | string;
-
 	private deletedAt: null | string;
-
 	private email: string;
-
+	private groups: Array<
+		{
+			permissions: Array<Pick<PermissionModel, "key" | "name">>;
+		} & Pick<GroupModel, "name">
+	>;
```

Comment:
I see that in other places we use model in entity, i think we will need to change it to entity. this way we can do it like this
```suggestion
	private groups: Pick<GroupEntity, "name" | "permissions">[]
```

Reply:
It doesn't work, I tried to do it that way, it doesn't work.